lib/galaxy/tools/evaluation.py

```diff
@@ -47,6 +51,48 @@
 
 log = logging.getLogger(__name__)
 
+PLAN = """
+Create job prolog script,
+- runs metadata script on inputs if necessary
+  - stages remote inputs
+- builds command line script
+  - takes as input:
+    - Job inputs / outputs, parameters
+    - Serialized tool ?
+- read command line back in set_metadata script
+
+======
+AND/OR
+======
+
+Schedule remote celery worker with one-time job handler, could do staging ... but yikes, I don't think that's atomic in any way
+Probably better to implement the above thing, then turn it into celery tasks for faster startup / less dependencies on worker.
+
+
+Problems:
+Do we want to build a full app ? Probably not.
+Do we want to supply galaxy config file ? Probably not.
+Dataset conversions ... would need to move outside of ToolEvaluator
```

Comment:
I'm going to take a look at this TODO today - I'll report back if I make any progress on determining if I think it is something I could plausibly help with.

Reply:
I have some more work locally, just didn't get around to pushing and updating this here.
I haven't solved / thought about the dataset conversions yet, but I think most of the rest I have figured out.
I'm positive I can wrap this up once we're done with the release process / the obvious bugs are worked out.

I think in the overall picture it would be good to work out how a deferred dataset would look like in the database and how the UI could look like, any chance you could take a look at this ?